src/libcephsqlite.cc

```diff
@@ -0,0 +1,927 @@
+  }
+
+  ret = io_ctx->create(get_lock_file_name(db_file, SQLITE_LOCK_RESERVED), must_create);
+  if (ret < 0 && ret != -EEXIST) {
+    // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::error: while creating:" << get_lock_file_name(db_file, SQLITE_LOCK_RESERVED) << std::endl;
+    return SQLITE_ERROR;
+  }
+
+  ret = io_ctx->create(get_lock_file_name(db_file, SQLITE_LOCK_PENDING), must_create);
+  if (ret < 0 && ret != -EEXIST) {
+    // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::error: while creating:" << get_lock_file_name(db_file, SQLITE_LOCK_PENDING) << std::endl;
+    return SQLITE_ERROR;
+  }
+
+  ret = io_ctx->create(get_lock_file_name(db_file, SQLITE_LOCK_EXCLUSIVE), must_create);
+  if (ret < 0 && ret != -EEXIST) {
+    // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::error: while creating:" << get_lock_file_name(db_file, SQLITE_LOCK_EXCLUSIVE) << std::endl;
+    return SQLITE_ERROR;
+  }
+
+  return SQLITE_OK;
+}
+
+static
+int remove_lock_files(librados::IoCtx *io_ctx, const std::string &db_file)
+{
+  io_ctx->remove(get_lock_file_name(db_file, SQLITE_LOCK_SHARED));
+  io_ctx->remove(get_lock_file_name(db_file, SQLITE_LOCK_RESERVED));
+  io_ctx->remove(get_lock_file_name(db_file, SQLITE_LOCK_PENDING));
+  io_ctx->remove(get_lock_file_name(db_file, SQLITE_LOCK_EXCLUSIVE));
+
+  return SQLITE_OK;
+}
+
+/* input is the full URI file name*/
+static
+librados::IoCtx *get_io_ctx(const char *zPath)
+{
+  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << "zPath:" << zPath << std::endl;
+
+  librados::IoCtx *io_ctx = nullptr;
+
+  vfs_context_map_mutex.lock();
+  if (vfs_context_map[std::string(zPath)])
+    io_ctx = vfs_context_map[std::string(zPath)]->io_ctx;
+  vfs_context_map_mutex.unlock();
+
+  return io_ctx;
+}
+
+/* input is the full URI file name*/
+static
+libradosstriper::RadosStriper *get_radosstriper(const char *zPath)
+{
+  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << "zPath:" << zPath << std::endl;
+
+  libradosstriper::RadosStriper *rs = nullptr;
+
+  vfs_context_map_mutex.lock();
+  if (vfs_context_map[std::string(zPath)])
+    rs = vfs_context_map[std::string(zPath)]->rs;
+  vfs_context_map_mutex.unlock();
+
+  return rs;
+}
+
+static
+std::string get_db_name(const char *zName)
+{
+  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << "zName:" << zName << std::endl;
+  /* eg. "accounts.db-journal"
+   * so filename should be extracted as "accounts.db"
+   * NOTE main database file name will always have a .db extension
+   *      see ceph_sqlite3_open()
+   */
+  const char *e = strstr(zName, ".db-");
+  if (!e)
+      return std::string(zName);
+
+  e += 3; // point to the '-' after .db
+
+  return std::string(zName, e - zName);
+}
+
+static
+int get_lock_type(const std::string &db_name, const std::string &file_name)
+{
+  int eLock = SQLITE_LOCK_NONE;
```

Comment:
we don't use camelCase naming convention. could you just use `lock_type` instead?